src/matrixrtc/MembershipManager.ts

```diff
@@ -11,19 +11,46 @@ import { Focus } from "./focus.ts";
 import { isLivekitFocusActive } from "./LivekitFocus.ts";
 import { MembershipConfig } from "./MatrixRTCSession.ts";
 
+export abstract class MembershipManagerInterface {
+    public constructor(
+        joinConfig: MembershipConfig | undefined,
+        room: Pick<Room, "getLiveTimeline">,
+        client: Pick<
+            MatrixClient,
+            | "getUserId"
+            | "getDeviceId"
+            | "sendStateEvent"
+            | "_unstable_sendDelayedEvent"
+            | "_unstable_sendDelayedStateEvent"
+        >,
+        getOldestMembership: () => CallMembership | undefined,
+    ) {}
+    public abstract isJoined(): boolean;
+    public abstract join(fociPreferred: Focus[], fociActive?: Focus): void;
+    public abstract leave(timeout: number | undefined): Promise<boolean>;
+    public abstract onMembershipsUpdate(): Promise<void>;
+    public abstract getActiveFocus(): Focus | undefined;
+}
+
 /**
- *  This internal class is used by the MatrixRTCSession to manage the local user's own membership of the session.
+ * This internal class is used by the MatrixRTCSession to manage the local user's own membership of the session.
+ *
+ * Its responsibitiy is to manage the locals user membership:
+ *  - send that sate event
+ *  - send the delayed leave event
+ *  - update the delayed leave event while connected
+ *  - update the state event when it times out (for calls longer than membershipExpiryTimeout ~ 4h)
+ *
+ * It is possible to test this class on its own. The api surface (to use for tests) is
+ * defined in `MembershipManagerInterface`.
+ *
+ * It is recommended to only use this interface for testing to allow replacing this class.
+ *
  *  @internal
  */
 export class MembershipManager {
```

Comment:
This should presumably extend `MembershipManagerInterface`

Reply:
since the super class is all abstract and calling `super()` is a noop I opted to `implement` instead of `extend`